File: homework/hw1/python/queens_bfs_dfs.py
from time import time

from algorithm.breadth_first_search import BreadthFirstSearch
from algorithm.depth_first_search import DepthFirstSearch
from problem.queens import QueensState

if __name__ == '__main__':    
    
    for i in range(8,14):
        t0 = time()
        
        s = QueensState(i)
        
        bfs = BreadthFirstSearch(s)
        
        bfs.search(True, False)
        
        t1=time()

        print(f"bfs: time = {t1 - t0}s")

        dfs = DepthFirstSearch(s)
        
        dfs.search(True, False)
        
        print(f"dfs: time = {time() - t1}s")

File: homework/hw1/c++/queens_bfs_dfs.cpp
#include <iostream>
#include <ctime>

#include "problem/queens.hpp"

#include "algorithm/depth_first_search.hpp"
#include "algorithm/breadth_first_search.hpp"

int main(){
    std::ios::sync_with_stdio(false);
    for(int i=8;i<=15;i++)
    {
    time_t t0 = time(nullptr);
    
    QueensState state(i);
    BreadthFirstSearch<QueensState> bfs(state);
    bfs.search(true, false);
    time_t t1 = time(nullptr);
    DepthFirstSearch<QueensState> dfs(state);
    dfs.search(true, false);
    
    std::cout << "bfs: "<< t1-t0 << std::endl;
    std::cout << "dfs: "<< time(nullptr) - t1 << std::endl;
    }
    return 0;
}
